lib/finch.ex

```diff
@@ -131,6 +161,56 @@ defmodule Finch do
     end
   end
 
+  def pool_options!(pools) do
+    Enum.reduce(pools, %{default: default_pool_options()}, fn {destination, opts}, acc ->
+      case cast_pool_opts(opts) do
+        {:ok, validated} ->
+          Map.put(acc, destination, validated)
+
+        {:error, reason} ->
+          raise ArgumentError, "got invalid configuration for pool #{destination}! #{reason}"
+      end
+    end)
+  end
+
+  defp cast_pool_opts(opts) do
+    with {:ok, size} <- validate_pool_size(opts),
+         {:ok, count} <- validate_pool_count(opts),
+         {:ok, backoff} <- validate_backoff(opts[:backoff]) do
+      {:ok, %{size: size, count: count, backoff: backoff, conn_opts: []}}
+    end
+  end
+
+  defp validate_pool_size(opts) do
+    opts |> Keyword.get(:size, @default_pool_size) |> validate_pos_integer()
+  end
+
+  defp validate_pool_count(opts) do
+    opts |> Keyword.get(:count, @default_pool_size) |> validate_pos_integer()
+  end
+
+  defp validate_backoff(nil) do
+    {:ok, %{initial: @default_backoff_initial, max: @default_backoff_max}}
+  end
+
+  defp validate_backoff(opts) do
+    init = Keyword.get(opts, :initial, @default_backoff_initial)
+    max = Keyword.get(opts, :initial, @default_backoff_max)
```

Comment:
This should be `max` here right?

Reply:
ah of course

Reply:
what do you think of `:initial`? Considering changing it to `:min`

Reply:
Hmm..I think either are fine. I kinda like `:initial` since that describes how we're using that specific parameter

Reply:
cool, I'll keep it as it!